omod/src/main/java/org/openmrs/module/idgen/rest/resource/LogEntryResource.java

```diff
@@ -0,0 +1,146 @@
+				"1.12.*", "2.0.*", "2.1.*" })
+public class LogEntryResource extends DelegatingCrudResource<LogEntry> {
+
+	@Override
+	protected NeedsPaging<LogEntry> doGetAll(RequestContext context) {
+		return new NeedsPaging<LogEntry>(
+				Context.getService(IdentifierSourceService.class).getLogEntries(null, null, null, null, null, null),
+				context);
+	}
+
+	@Override
+	protected PageableResult doSearch(RequestContext context) {
+		IdentifierSourceService identifierSourceService = Context.getService(IdentifierSourceService.class);
+		UserService service = Context.getUserService();
+
+		String source = context.getRequest().getParameter("source");
+		String fromDate = context.getRequest().getParameter("fromDate");
+		String toDate = context.getRequest().getParameter("toDate");
+		String identifier = context.getRequest().getParameter("identifier");
+		String comment = context.getRequest().getParameter("comment");
+		String generatedBy = context.getRequest().getParameter("generatedBy");
+
+		IdentifierSource logSource = source != null ? identifierSourceService.getIdentifierSourceByUuid(source) : null;
+		Date dateFrom = fromDate != null ? (Date) ConversionUtil.convert(fromDate, Date.class) : null;
+		Date dateTo = toDate != null ? (Date) ConversionUtil.convert(toDate, Date.class) : null;
+		User user = generatedBy != null ? service.getUserByUuid(generatedBy) : null;
+		if (logSource == null && identifier == null && comment == null && dateFrom == null && dateTo == null
+				&& user == null) {
+			return new EmptySearchResult();
+		}
+		List<LogEntry> logEntries = identifierSourceService.getLogEntries(logSource, dateFrom, dateTo, identifier, user,
+				comment);
+		return new NeedsPaging<LogEntry>(logEntries, context);
+	}
+
+	@Override
+	public DelegatingResourceDescription getRepresentationDescription(Representation rep) {
+		DelegatingResourceDescription description = null;
+		if (rep instanceof RefRepresentation) {
+			description = new DelegatingResourceDescription();
+			description.addProperty("uuid");
+			description.addProperty("display");
+			description.addProperty("identifier");
+			description.addSelfLink();
+		} else if (rep instanceof DefaultRepresentation || rep instanceof FullRepresentation) {
+			description = new DelegatingResourceDescription();
+			description.addProperty("uuid");
+			description.addProperty("source");
+			description.addProperty("identifier");
+			description.addProperty("comment");
+			description.addProperty("generatedBy");
+			description.addProperty("dateGenerated");
+			description.addSelfLink();
+			if (rep instanceof DefaultRepresentation) {
+				description.addLink("full", ".?v=" + RestConstants.REPRESENTATION_FULL);
+			}
+		}
+		return description;
+	}
+
+	@PropertyGetter("source")
+	public String getSource(LogEntry logentry) {
+		return logentry.getSource().getUuid();
+	}
+
+	@PropertyGetter("generatedBy")
+	public String getGeneratedBy(LogEntry logentry) {
+		return logentry.getGeneratedBy().getUuid();
```

Comment:
You do not need this method. Just change the webservices module dependency version to 2.13

Reply:
Ive changed the dependencies from 2.0 to 2.3

Reply:
to 2 point what? ☺️

Reply:
haha 2.13 not ~~2.3~~